### PR TITLE
chore(TDP-5280): cmf breadcrumb

### PR DESCRIPTION
**Pull request by charlesnguyen**
**Link to the JIRA issue**
https://jira.talendforge.org/browse/TDP-5280

**Please check if the PR fulfills these requirements**
- [ ] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] The code coverage on new code is > 75 % for backend and > 95% for frontend
- [ ] The new code does not introduce new technical issues (sonar / eslint)
- [ ] Functional tests have been performed
- [ ] Docker configuration files for config-std or config-cloud profiles are impacted

**Please check the browsers you've tested on**
- [ ] Chrome, Firefox, Safari, Edge, IE11
- [ ] No, that's bad, this PR should not be merged !
- [ ] No, and no need to (backend changes only)

### CODE REVIEW

File: dataprep-webapp/src/app/next/sagas/effects/preparation.effects.js [hunk 1]
```diff
@@ -64,3 +67,17 @@ export function* setTitleEditionMode(payload) {
 export function* openAbout() {
 	yield put(actions.components.mergeState('PreparationCreatorModal', 'default', { show: true }));
 }
+
+
+export function* fetchFolder(payload) {
+	const defaultFolderId = 'Lw==';
+	const currentFolderPromise = yield put(
+		actions.http.get(`/api/folders/${payload.folderId || defaultFolderId}`, {
```
Comment (frassinier) [inline dataprep-webapp/src/app/next/sagas/effects/preparation.effects.js]: Prefer http-saga here

File: dataprep-webapp/src/app/next/services/preparation.service.js [hunk 2]
```diff
@@ -20,6 +20,21 @@ function transform({ folders = [], preparations = [] }) {
...
 
+function transformFolder({ folder, hierarchy }) {
+	const breadcrumbs = [
+		...hierarchy,
+		folder,
+	].map(folder =>
+		({
+			id: folder.id,
+			text: folder.name || 'Home',
+			title: folder.name || 'Home',
```
Comment (frassinier) [inline dataprep-webapp/src/app/next/services/preparation.service.js]: it's not really i18n ready

File: dataprep-webapp/src/app/next/services/preparation.service.js [hunk 3]
```diff
@@ -20,6 +20,21 @@ function transform({ folders = [], preparations = [] }) {
...
+		}));
+	return breadcrumbs;
```
Comment (frassinier) [inline dataprep-webapp/src/app/next/services/preparation.service.js]: return map fn directly

File: dataprep-webapp/src/app/next/sagas/effects/preparation.effects.js [hunk 4]
```diff
@@ -27,14 +29,10 @@ export function* duplicate(prep) {
...
-	yield put(
-		actions.http.get(`/api/folders/${payload.folderId || defaultFolderId}/preparations`, {
-			cmf: {
-				collectionId: 'preparations',
-			},
-			transform: PreparationService.transform,
-		}),
-	);
+	const uris = yield select(state => state.cmf.collections.getIn(['settings', 'uris']));
+	const { data } = yield call(http.get, `${uris.get('apiFolders')}/${(payload.folderId || defaultFolderId)}/preparations`);
```
Comment (ncomont) [inline dataprep-webapp/src/app/next/sagas/effects/preparation.effects.js]: Please fix your prettier conf

File: dataprep-webapp/src/app/next/sagas/effects/preparation.effects.js [hunk 5]
```diff
@@ -27,14 +29,10 @@ export function* duplicate(prep) {
...
+	yield put(actions.collections.addOrReplace('preparations', PreparationService.transform(JSON.parse(data))));
```
Comment (ncomont) [inline dataprep-webapp/src/app/next/sagas/effects/preparation.effects.js]: Why do you need a `JSON.parse` now ?

File: dataprep-webapp/src/app/next/sagas/effects/preparation.effects.js [hunk 6]
```diff
@@ -64,3 +62,12 @@ export function* setTitleEditionMode(payload) {
 export function* openAbout() {
 	yield put(actions.components.mergeState('PreparationCreatorModal', 'default', { show: true }));
 }
+
+export function* fetchFolder(payload) {
+	const defaultFolderId = 'Lw==';
+	const uris = yield select(state => state.cmf.collections.getIn(['settings', 'uris']));
+	const { data } = yield call(http.get, `${uris.get('apiFolders')}/${(payload.folderId || defaultFolderId)}`);
+	yield put(actions.components.mergeState('Breadcrumbs', 'default', new Map({
+		items: PreparationService.transformFolder(JSON.parse(data)),
```
Comment (ncomont) [inline dataprep-webapp/src/app/next/sagas/effects/preparation.effects.js]: Same question

File: dataprep-webapp/src/app/next/components/Breadcrumbs/Breadcrumbs.connect.js [hunk 7]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
```
Comment (ncomont) [inline dataprep-webapp/src/app/next/components/Breadcrumbs/Breadcrumbs.connect.js]: Please put this connect in the monorepo

File: dataprep-webapp/src/app/next/sagas/effects/preparation.effects.js (hunk 5 above)
Comment (charlesnguyen) [reply dataprep-webapp/src/app/next/sagas/effects/preparation.effects.js]: To be sur we have a object to update the cmf collection

File: dataprep-webapp/src/app/next/sagas/effects/preparation.effects.js (hunk 6 above)
Comment (charlesnguyen) [reply dataprep-webapp/src/app/next/sagas/effects/preparation.effects.js]: To be sur we have a object to update the cmf collection

File: dataprep-webapp/src/app/next/components/Breadcrumbs/Breadcrumbs.connect.js (hunk 7 above)
Comment (charlesnguyen) [reply dataprep-webapp/src/app/next/components/Breadcrumbs/Breadcrumbs.connect.js]: https://github.com/Talend/ui/pull/1584

File: dataprep-webapp/src/app/configure.cmf.js [hunk 8]
```diff
@@ -25,7 +25,7 @@ const registerRouteFunction = api.route.registerFunction;
  * - Register your components in the CMF dictionary
  * - Register action creators in CMF actions dictionary
  */
-export default function initialize(additionalConfiguration = {}) {
+export default function initialize(appSettings, additionalConfiguration = {}) {
```
Comment (ncomont) [inline dataprep-webapp/src/app/configure.cmf.js]: please revert that

File: dataprep-webapp/src/app/next/sagas/effects/preparation.effects.js (hunk 6 above)
Comment (ncomont) [reply dataprep-webapp/src/app/next/sagas/effects/preparation.effects.js]: please fix the content type emitted instead

**Pull request merged**